### PR TITLE
Refactor contains function such that it does a simpler query to deter…

### PR DESCRIPTION
**Pull request by seamuslee001**
…mine if geometries are within each other

### CODE REVIEW

File: CRM/CiviGeometry/BAO/Geometry.php [hunk 1]
```diff
@@ -87,57 +87,61 @@ public static function getCentroid($params) {
...
+      while ($res->fetch()) {
+        if ($res->contains_result) {
+          $multipleResult[] = $res->id;
         }
       }
       return $multipleResult;
     }
     if (is_numeric($params['geometry_b'])) {
-      $sql = $duleIntegerSQL;
+      $sql = $duleIntegerSQL . " AND a.id = %1";
```
Comment (johntwyman) [inline CRM/CiviGeometry/BAO/Geometry.php]: Is this meant to be "dual"?

File: CRM/CiviGeometry/BAO/Geometry.php [hunk 2]
```diff
@@ -87,57 +87,61 @@ public static function getCentroid($params) {
    */
   public static function contains($params) {
     $multipleResult = [];
-    $duleIntegerSQL = "SELECT ST_Contains(a.geometry, b.geometry)
+    $duleIntegerSQL = "SELECT a.id, ST_Contains(a.geometry, b.geometry) as contains_result
```
Comment (johntwyman) [inline CRM/CiviGeometry/BAO/Geometry.php]: "dual"

File: CRM/CiviGeometry/BAO/Geometry.php (hunk 1 above)
Comment (seamuslee001) [reply CRM/CiviGeometry/BAO/Geometry.php]: fixed

File: CRM/CiviGeometry/BAO/Geometry.php (hunk 2 above)
Comment (seamuslee001) [reply CRM/CiviGeometry/BAO/Geometry.php]: fixed

File: CRM/CiviGeometry/BAO/Geometry.php [hunk 3]
```diff
@@ -87,57 +87,61 @@ public static function getCentroid($params) {
...
-          $res = CRM_Core_DAO::singleValueQuery($singleIntergerSQL, [
-            1 => [$params['geometry_b'], 'String'],
-            2 => [$geometry['id'], 'Positive'],
-          ]);
-        }
-        if (!empty($res)) {
-          $multipleResult[] = $geometry['id'];
+      $geometries = civicrm_api3('Geometry', 'get', $geometryParams)['values'];
+      $singleIntergerSQL .= " WHERE id IN (%2)";
+      $dualIntegerSQL .= "AND a.id IN (%1)";
```
Comment (andrew-cormick-dockery) [inline CRM/CiviGeometry/BAO/Geometry.php]: I think you're going to need an extra space before "AND" here.

File: CRM/CiviGeometry/BAO/Geometry.php [hunk 4]
```diff
@@ -87,57 +87,61 @@ public static function getCentroid($params) {
...
+      if (is_numeric($params['geometry_b'])) {
+        $res = CRM_Core_DAO::executeQuery($dualIntegerSQL, [
+          1 => [implode(', ', $geoemtry_ids), 'CommaSeparatedIntegers'],
+          2 => [$params['geometry_b'], 'Positive'],
+        ]);
+      }
+      else {
+        $res = CRM_Core_DAO::executeQuery($singleIntergerSQL, [
+          1 => [$params['geometry_b'], 'String'],
+          2 => [implode(', ', $geoemtry_ids), 'CommaSeparatedIntegers'],
```
Comment (andrew-cormick-dockery) [inline CRM/CiviGeometry/BAO/Geometry.php]: sp geometry

Comment (seamuslee001) [reply CRM/CiviGeometry/BAO/Geometry.php]: Fixed @andrew-cormick-dockery

File: CRM/CiviGeometry/BAO/Geometry.php (hunk 3 above)
Comment (seamuslee001) [reply CRM/CiviGeometry/BAO/Geometry.php]: @andrew-cormick-dockery good pick up i extended a unit test in https://github.com/australiangreens/au.org.greens.civigeometry/pull/12/commits/c530d86541581cd6e7104b97492acaf18285bc05 which demonstrates the issue you raise here and then fixed it in https://github.com/australiangreens/au.org.greens.civigeometry/pull/12/commits/25e3bac79c14248ce39c276f81dee9298537724d

Comment (seamuslee001) [reply CRM/CiviGeometry/BAO/Geometry.php]: Test failure demonstrated here https://buildkite.com/greens/au-dot-org-dot-greens-dot-civigeometry/builds/52#768d9204-f68d-4515-a385-359761b78eef/942-970 and passing here https://buildkite.com/greens/au-dot-org-dot-greens-dot-civigeometry/builds/53#5c4d0364-28bc-408d-b955-f9f3d0397451/769-795

**Pull request merged**